hledger-lib/Hledger/Read/Common.hs

```diff
@@ -1011,17 +1001,15 @@ data AmbiguousNumber = AmbiguousNumber DigitGrp Char DigitGrp
 -- | Description of a single digit group in a number literal.
 -- "Thousands" is one well known digit grouping, but there are others.
 data DigitGrp = DigitGrp {
-  digitGroupLength :: !Int,    -- ^ The number of digits in this group.
-  digitGroupNumber :: !Integer -- ^ The natural number formed by this group's digits.
+  digitGroupLength :: !Word,    -- ^ The number of digits in this group.
+  digitGroupNumber :: !Integer  -- ^ The natural number formed by this group's digits.
```

Comment:
since we're changing things.. is there a better type for this ? we don't need/want this to be negative

Reply:
Hmm. Since it can have up to 255 digits, even `Word` wouldn't be big enough, so it would have to be `Natural`. I'm not sure it's worth the change, but we could.